Development/nmos/streamcompatibility_behaviour.cpp

```diff
@@ -0,0 +1,306 @@
+            }));
+        }
+
+        nmos::sender_state validate_sender_resources(const web::json::value& transport_file, const web::json::value& flow, const web::json::value& source, const web::json::array& constraint_sets)
+        {
+            nmos::sender_state sender_state;
+
+            if (!web::json::empty(constraint_sets))
+            {
+                bool constrained = true;
+
+                auto source_found = std::find_if(constraint_sets.begin(), constraint_sets.end(), [&](const web::json::value& constraint_set) { return match_source_parameters_constraint_set(source, constraint_set); });
+                auto flow_found = std::find_if(constraint_sets.begin(), constraint_sets.end(), [&](const web::json::value& constraint_set) { return match_flow_parameters_constraint_set(flow, constraint_set); });
+
+                constrained = constraint_sets.end() != source_found && constraint_sets.end() != flow_found;
+
+                if (!transport_file.is_null() && !transport_file.as_object().empty())
+                {
+                    utility::string_t sdp_data = get_sdp_data(transport_file);
+                    const auto session_description = sdp::parse_session_description(utility::us2s(sdp_data));
+                    auto sdp_params = nmos::parse_session_description(session_description).first;
+
+                    const auto format_params = nmos::details::get_format_parameters(sdp_params);
+                    const auto sdp_found = std::find_if(constraint_sets.begin(), constraint_sets.end(), [&](const web::json::value& constraint_set) { return nmos::details::match_sdp_parameters_constraint_set(nmos::details::format_constraints, sdp_params, format_params, constraint_set); });
+
+                    constrained = constrained && constraint_sets.end() != sdp_found;
+                }
+
+                sender_state = constrained ? nmos::sender_states::constrained : nmos::sender_states::active_constraints_violation;
+            }
+            else
+            {
+                sender_state = nmos::sender_states::unconstrained;
+            }
+
+            return sender_state;
+        }
+
+        nmos::receiver_state validate_receiver_resources(const web::json::value& transport_file, const web::json::value& receiver)
+        {
+            nmos::receiver_state receiver_state;
+
+            if (!transport_file.is_null() && !transport_file.as_object().empty())
+            {
+                utility::string_t sdp_data = get_sdp_data(transport_file);
+                const auto session_description = sdp::parse_session_description(utility::us2s(sdp_data));
+                auto sdp_params = nmos::parse_session_description(session_description).first;
+
+                receiver_state = nmos::receiver_states::compliant_stream;
+
+                try
+                {
+                    validate_sdp_parameters(receiver, sdp_params);
+                }
+                catch (const std::runtime_error& e)
```

Comment:
Yes, IS-11 dictates an implementation server to give the reason for the validation failure in `debug` property for each `status` resource (https://specs.amwa.tv/is-11/branches/v1.0-dev/docs/Behaviour_-_Server_Side.html#status-of-input). Fixed it.

Regarding logging, it would be consistent to log any validation result. Should I implement it?

Reply:
Included validation failure reasons in the log.